schema/v1/trigger-event.json

```diff
@@ -0,0 +1,188 @@
+    },
+    "eventClassName": {
+      "type": "string",
+      "description": "A fully qualified class name for the trigger event."
+    },
+    "eventBusinessName": {
+      "type": "string",
+      "description": "A business name for the trigger event."
+    },
+    "recipientType": {
+      "$ref": "#/$defs/RecipientType"
+    },
+    "extractor": {
+      "type": "string",
+      "description": "The SpEL expression to extract a recipient or subscription."
+    },
+    "subscriptionType": {
+      "$ref": "#/$defs/SubscriptionType"
+    },
+    "eventSlicer": {
+      "type": "string",
+      "description": "The SpEL expression to process entries separately in case the event is an array."
+    },
+    "serviceAgreementExtractor": {
+      "type": "string",
+      "description": "The SpEL expression to extract a service agreement."
+    },
+    "generalNotifications": {
+      "type": "array",
+      "description": "One or more general notifications.",
+      "items": {
+        "$ref": "#/$defs/GeneralNotification"
+      }
+    }
+  },
+  "required": [
+    "id"
+  ],
+  "additionalProperties": false,
+  "$defs": {
+    "RecipientType": {
+      "type": "string",
+      "description": "Identifies the type of recipient based on their subscription",
+      "enum": [
+        "subscription",
+        "internal"
+      ]
+    },
+    "SubscriptionType": {
+      "type": "string",
+      "description": "The entity type if recipientType is subscription.",
+      "enum": [
+        "arrangement"
+      ]
+    },
+    "DeliveryOptions": {
+      "type": "object",
+      "title": "Delivery Option",
+      "properties": {
+        "channel": {
+          "type": "string",
+          "description": "A delivery channel."
+        },
+        "channelSettings": {
+          "type": "string",
+          "description": "A path to channel-specific settings that should be present in the import package."
+        },
+        "isEnabledByDefault": {
+          "type": "boolean",
+          "description": "Indicates whether the channel is enabled by default."
+        },
+        "isManageableByUser": {
+          "type": "boolean",
+          "description": "Indicates whether the user can manage preferences for the channel."
+        },
+        "routeInfo": {
+          "type": "object",
+          "description": "A key for mapping to the route of the web and mobile application."
+        },
+        "engagementTemplates": {
+          "type": "array",
+          "description": "One or more engagement templates.",
+          "items": {
+            "$ref": "#/$defs/EngagementTemplates"
+          }
+        }
+      },
+      "additionalProperties": false
+    },
+    "GeneralNotification": {
+      "type": "object",
+      "title": "General Notification",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The unique identifier of the general notification."
+        },
+        "name": {
```

Comment:
@eacarvalho why an `id` and `name` is the name supposed to be a friendly name or something displayed on engage app?